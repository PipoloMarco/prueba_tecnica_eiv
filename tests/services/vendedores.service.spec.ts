import { HttpClientModule } from '@angular/common/http';
import { TestBed } from '@angular/core/testing';
import { VendedoresService } from '../../src/app/services/vendedores.service';
import { of } from 'rxjs';

describe('Vendedores Services', () => {
  let service: VendedoresService;

  const mockVendedores = [
    {
      id: 1,
      usuarioLogin: 'marcoo',
      nombre: 'Perez, Anibal ',
      habilitado: true,
      fechaNacimiento: '1982-05-01',
      observaciones: null,
      localidad: {
        id: 1,
        localidad: 'Rosario',
        codigoPostal: '2000',
      },
    },
  ];

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientModule],
    });
    service = TestBed.inject(VendedoresService);
  });

  test('Se crea el servicio', () => {
    expect(service).toBeTruthy();
  });

  test('debe de traer informacion del vendedor', (done) => {
    // Mock el servicio para que retorne el array de vendedores mockeados en vez de hacer la peticiÃ³n real
    jest.spyOn(service, 'getVendedores').mockReturnValue(of(mockVendedores));
    service.getVendedores().subscribe((vendedores) => {
      console.log(vendedores);
      expect(vendedores).toEqual(mockVendedores);
      expect(vendedores.length).toBeGreaterThan(0);
      expect(vendedores[0].nombre).toBe('Perez, Anibal ');
      done();
    });
  });

  test('debe devolver objetos con la estructura correcta', (done) => {
    jest.spyOn(service, 'getVendedores').mockReturnValue(of(mockVendedores));
    service.getVendedores().subscribe((vendedores) => {
      vendedores.forEach((vendedor) => {
        expect(vendedores).toEqual(mockVendedores);
        expect(vendedor).toHaveProperty('id');
        expect(vendedor).toHaveProperty('usuarioLogin');
        expect(vendedor).toHaveProperty('nombre');
        expect(vendedor).toHaveProperty('habilitado');
        expect(vendedor).toHaveProperty('observaciones');
        expect(vendedor).toHaveProperty('localidad');
      });
      done();
    });
  });

  test('Debe de traer la foto', () => {});
});
// "jest": {
//   "preset": "jest-preset-angular",
//   "setupFilesAfterEnv": [
//     "<rootDir>/setup-jest.ts"
//   ],

//   "globalSetup": "jest-preset-angular/global-setup"
// }
