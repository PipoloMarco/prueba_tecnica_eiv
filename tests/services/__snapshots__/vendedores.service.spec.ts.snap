// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vendedores Services debe de traer informacion del vendedor 1`] = `
[
  {
    "fechaNacimiento": "1982-05-01",
    "habilitado": true,
    "id": 1,
    "localidad": {
      "codigoPostal": "2000",
      "id": 1,
      "localidad": "Rosario",
    },
    "nombre": "Perez, Anibal ",
    "observaciones": null,
    "usuarioLogin": "marcoo",
  },
]
`;
